src/components/PricelistForm.js

```diff
@@ -1,12 +1,24 @@
-import React from "react";
-import { combine, withHistory, withModulesManager, Form } from "@openimis/fe-core";
+import React, { useEffect } from "react";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+
+import ReplayIcon from "@material-ui/icons/Replay";
+
+import { withHistory, withModulesManager, Form } from "@openimis/fe-core";
+import { clearMedicalPricelistItems } from "../actions";
 import PricelistGeneralPanel from "./PricelistGeneralPanel";
 import PricelistDetailsPanel from "./PricelistDetailsPanel";
-import ReplayIcon from "@material-ui/icons/Replay";
 
 const PricelistForm = (props) => {
-  const { readOnly, onBack, onSave, onReset, pricelist, onChange, fetchDetails, details } = props;
-  const canSave = () => pricelist.name && pricelist.pricelistDate;
+  const { readOnly, onBack, onSave, onReset, pricelist, onChange, fetchDetails, details, isValid, clearMedicalPricelistItems} = props;
+  const canSave = () => pricelist.name && pricelist.pricelistDate && isValid === true;
+
+  useEffect(() => {
+    return () => { 
+      clearMedicalPricelistItems();
```

Comment:
This seems to be working for both items and services, so why Items in the name?

Reply:
That's a good point, it could be misleading. Name changed to _clearMedicalPricelists_, thanks @wzglinieckisoldevelo .